src/main/java/pl/javastart/restoffers/offers/OffersRestController.java

```diff
@@ -0,0 +1,55 @@
+package pl.javastart.restoffers.offers;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RequestMapping("api/offers")
+@RestController
+public class OffersRestController {
+
+    private final OffersService offersService;
+
+    public OffersRestController(OffersService offersService) {
+        this.offersService = offersService;
+    }
+
+    @GetMapping
+    List<Offer> getAllOffers(@RequestParam(name = "title", required = false) String text) {
+        if (text != null) {
+            return offersService.findOffers(text);
+        } else {
+            return offersService.findAll();
+        }
+    }
+
+    @GetMapping("/count")
+    int countOffers() {
+        return offersService.findAll().size();
```

Comment:
Zamiast pobierać obiekty przez findAll(), wystarczyłoby  wywołać count() na repozytorium. W przypadku dużej liczby obiektów to ogromna oszczędność w ilości pobieranych danych z bazy